diff --git a/setup.py b/setup.py
index a7f848c..c66fa23 100644
--- a/setup.py
+++ b/setup.py
@@ -126,7 +126,7 @@ class CMakeBuild(build_ext):
             build_args += ["--", "/m"]
         else:
             cmake_args += ["-DCMAKE_BUILD_TYPE=" + cfg]
-            build_args += ["--", "-j2"]
+            build_args += ["--", f"-j{os.cpu_count()}"]

         env = os.environ.copy()
         env["CXXFLAGS"] = '{} {} -DVERSION_INFO=\\"{}\\"'.format(
